tests/unit/providers/common/outbound_filter_tests.py

```diff
@@ -0,0 +1,101 @@
+import unittest
+import pytest
+
+
+from rbac.providers.common.outbound_filters import (
+    outbound_group_creation_filter,
+    outbound_group_filter,
+    outbound_user_filter,
+    outbound_user_creation_filter,
+)
+
+LIST_OF_TEST_USERS = [
+    {
+        "account_enabled": True,
+        "name": "test_user",
+        "user_principal_name": "test_user@test_mail.com",
+    },
+    {"account_enabled": True},
+    {"group_nickname": "test_group"},
+    {"name": "test_user", "user_principal_name": "test_user@test_mail.com"},
+]
+
+
+def test_outbound_user_filter():
+    """ Test outbound user filter with valid user """
+    result = outbound_user_filter({"user_id": 1234}, "azure")
+    assert isinstance(result, dict) is True
+    assert result["id"] == 1234
+    assert "job_title" not in result
+
+
+def test_outbound_user_filter_bad_provider():
+    """ Test outbound user filter with bad provider throws error"""
+    with pytest.raises(TypeError):
+        outbound_user_filter({"user_id": 1234}, "test_run")
+
+
+def test_outbound_group_filter():
+    """ Test outbound group filter with valid user """
+    result = outbound_group_filter({"role_id": 1234}, "ldap")
+    assert result["objectGUID"] == 1234
+    assert "id" not in result
+
+
+def test_outbound_group_filter_bad_provider():
+    """ Test outbound group filter with bad provider throws error"""
+    with pytest.raises(TypeError):
+        outbound_group_filter({"user_id": 1234}, "test_run")
+
+
+def test_outbound_user_creation():
+    """ Test outbound user creation with valid provider and account"""
+    user = LIST_OF_TEST_USERS[0]
+    result = outbound_user_creation_filter(user, "azure")
+    assert result["accountEnabled"] is True
+    assert result["mailNickname"] == "test_user"
+    assert "mail" not in result
+
+
+def test_outbound_user_creation_without_accountEnabled():
+    """ Test outbound user creation without accountEnabled passed in"""
+    user = LIST_OF_TEST_USERS[3]
+    result = outbound_user_creation_filter(user, "azure")
+    assert result["accountEnabled"] is True
+
+
+def test_outbound_user_creation_with_bad_provider():
+    """ Test outbound group creation with bad provider throws error"""
+    user = LIST_OF_TEST_USERS[1]
```

Comment:
So when you are running though multiple test cases you want to use the parameterize function of pytest.  So it is more for vaulting through multiple inputs in a test rapidly rather then doing it via assignment like this.  Otherwise you would just do it via assignment.  So we should modify this.

Reply:
I think I'm using the parameterize function correctly. Please let me know if you see further changes that are needed!